### PR TITLE
#167749949 JWT authentication

### PR DESCRIPTION
**Pull request by hustlaviola**
#### What does this PR do?

This PR implements JWT authentication and add it to user response on registration

#### Description of Tasks to be completed

- Update the test to have `token` property
- Remove `session` snippets from the codebase
- Create a function to generate a token
- Create a function to authenticate the user
- Update signup function to return token
- Update Swagger documentation with Security Definitions

#### How should this be manually tested?

- Clone the Repo, change into the directory and run `npm install`
- Run `npm dev` to start the server
- Make a `POST` request to `http://localhost:{PORT}/api/v1/user/signup` on Postman using valid credentials

Sample request body
```
{
    "email": "naimatjdavid@mail.com",
    "firstname": "Ibrahim",
    "lastname": "Jibson",
    "password": "ChibuokemAlabi99"
}
```

#### Relevant pivotal tracker story

[167749949](https://www.pivotaltracker.com/story/show/167749949)

#### Screenshot

Screenshot (129)

### CODE REVIEW

File: src/middlewares/userValidations.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import Responses from '../utils/Responses';
+import models from '../database/models';
+
+/**
+ * @function
+ * @description Validates user credentials
+ * @param {object} path - The signup schema
+ * @returns {object} JSON response
+ */
+const validateSignup = path => (req, res, next) => {
```
Comment (ldonjibson) [inline src/middlewares/userValidations.js]: as disccussed make it generic

**Review by ldonjibson — APPROVED** [review]
Weldone @hustlaviola

**Review by daviddamilola — CHANGES_REQUESTED** [review]
great work, however, the functionality of verifying token should be abstracted into its own function as it would be reused in other features you can name it `verifyToken(token)` and let it return the decoded object, it can belong to the helper class

**Comment by hustlaviola** [comment]
> great work, however, the functionality of verifying token should be abstracted into its own function as it would be reused in other features you can name it `verifyToken(token)` and let it return the decoded object, it can belong to the helper class

Feedback implemented @daviddamilola. Thank you

**Review by daviddamilola — APPROVED** [review]

File: src/utils/userSchema.js [hunk 2]
```diff
@@ -23,5 +23,9 @@ export default {
     lastName,
     email,
     password
+  }),
+  signin: Joi.object().keys({
+    email: Joi.string().required().label('Email is required'),
```
Comment (hustlaviola) [inline src/utils/userSchema.js]: I think it's necessary to validate this email, a user may input a number or a 2 letter word, we don't have to check the database to see if it exists. Let's use the schema provided above for that.
```suggestion
    email,
```

**Pull request merged**